Code/Framework/AzCore/AzCore/Prefab/PrefabEditorPreferences.cpp

```diff
@@ -37,4 +39,24 @@ namespace AzToolsFramework::Prefab
         return prefabOverridesUxEnabled;
     }
 
-} // namespace AzToolsFramework::Prefab
+    bool IsInspectorOverrideManagementEnabled()
+    {
+        bool isInspectorOverrideManagementEnabled = false;
+        bool prefabOverridesUxEnabled = false;
+
+        if (auto* registry = AZ::SettingsRegistry::Get())
+        {
+            registry->Get(isInspectorOverrideManagementEnabled, InspectorOverrideManagementKey);
+            registry->Get(prefabOverridesUxEnabled, EnablePrefabOverridesUxKey);
+        }
+
+        bool dpeEnabled = false;
+        if (auto* console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+        {
+            console->GetCvarValue("ed_enableDPE", dpeEnabled);
+        }
+
+        return (isInspectorOverrideManagementEnabled && prefabOverridesUxEnabled && dpeEnabled);
+    }
+
```

Comment:
Why was this moved to AzCore?

This is purely The PrefabEditorPreferences is purely related to a Tools workflow.

Reply:
I moved it here because there are some changes in AzCore/Component/* and AzFramework/DocumentPropertyEditor classes that are safer to wrap behind the feature flag until the overrides feature is fully ready. So I put just the helper function in AzCore to make that possible. I don't plan on adding any other prefab related code to AzCore, just the preferences.

Alternatively, I can move this to AzFramework too since nothing in AzCore needs this right now at least. Let me know your thoughts

Reply:
This function can be made as a lambda inside of the ReflectionAdapater.cpp `VisitObjectBegin` function if it is "temporary".

There is no reason to add the entire file Prefab file to AzCore.
Architectural wise, this shouldn't be in AzCore or AzFramework.

If a particular setting needs to be read temporary, then just use the string value directly `"/O3DE/Preferences/Prefabs/EnableInspectorOverrideManagement"`

Reply:
Yeah, I agree with the reasoning. I put the file in there since we can't reach AzToolsFramework from AzCore. But its probably an overkill and sets a bad precedent. I'll change it to query the setreg key directly